src/components/navigation/TopNavMenu.tsx

```diff
@@ -0,0 +1,97 @@
+import Button from '@mui/material/Button';
+import MenuItem from '@mui/material/MenuItem';
+import Menu from '@mui/material/Menu';
+import Link from '@mui/material/Link';
+import AccountCircleIcon from '@mui/icons-material/AccountCircle';
+import MenuRoundedIcon from '@mui/icons-material/MenuRounded';
+import Box from '@mui/material/Box';
+
+import { useCallback, useState } from 'react';
+import { Link as RouterLink } from 'react-router-dom';
+
+import useLogoutMutation from '../../api/mutations/account/useLogoutMutation';
+import useGetUserQuery from '../../api/queries/account/useGetUserQuery';
+import { mainNavigationStyles as styles } from './mainNavigation.styles';
+import { useAppSelector } from '../../hooks/redux-hooks';
+import { RoutesConfig } from '../../config/routes.config';
+
+export const TopNavMenu = () => {
+  const isLoggedIn = useAppSelector((state) => state.auth.token) !== null;
+  const userId = localStorage.getItem('sub');
+
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+  const open = Boolean(anchorEl);
+  const handleTouch = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const { isError } = useGetUserQuery(userId, isLoggedIn);
+  const { mutate } = useLogoutMutation();
+
+  const handleClose = useCallback(() => {
+    setAnchorEl(null);
+  }, []);
+  const handleLogout = () => mutate();
+
+  return (
+    <Box>
+      <Button
+        id="basic-button"
+        aria-controls={open ? 'basic-menu' : undefined}
+        aria-haspopup="true"
```

Comment:
not sure but should work without '="true"'

Reply:
it is related to a11y

don't know conscious or not, but it used correctly